### PR TITLE
Tenancy filterset updates

### LINKED ISSUE #1733: Add Missing Model Fields to Tenancy Filter Sets
See https://gist.github.com/jathanism/7c585954da4cb1a54997be31991cdaac#tenancy

### PR DESCRIPTION
**Pull request by glennmatthews**
# Closes: #1733 
# Closes: #2809
# What's Changed

- Rename `TenantFilterSet.tag` filter to `tags` (fixes #2803)
- Add missing filter fields to `TenantFilterSet` and `TenantGroupFilterSet` and remove redundant fields (fixes #1733)
- I didn't want to create yet another set of copy-paste boilerplate test data in `tenancy/tests/test_filters.py` so instead I created a bunch more basic factories for the models that weren't yet covered by our factories but were needed in this test.
    - I *have not* wired up these new factories to the `generate_test_data` management command yet as that would impact a tremendous number of other tests and inflate the scope of this PR to unmanageability. That can still be done as part of #1334 as time permits.
- Updated the migration documents, including some additional filterset changes from #1938 and #2849 that I noticed were inadvertently missed.

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- n/a Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- n/a Example Plugin Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Review by bryanculver — APPROVED** [review]
Very thorough. Thanks for the nice sprinkles of `TODO`s so we know to revisit.

**Review by HanlinMiao — APPROVED** [review]

File: nautobot/dcim/factory.py [hunk 1]
```diff
@@ -409,3 +419,90 @@ def parent(self):
...
+        "has_location", random_instance(lambda: Location.objects.get_for_model(VLANGroup), allow_null=False), None
+    )
+    site = factory.Maybe(
+        "has_location",
+        factory.LazyAttribute(lambda l: l.location.site or l.location.base_site),
+        random_instance(Site),
+    )
+
+    has_group = factory.Faker("pybool")
+    group = factory.Maybe("has_group", random_instance(RackGroup), None)  # TODO there's no RackGroupFactory yet...
```
[the remark this answers is not in the document]
Comment (glennmatthews) [reply nautobot/dcim/factory.py:458]: I don't need it for testing Tenants, and it's a field that's allowed to be null, so yeah I'd rather leave it out for now unless there's a strong need.
